pkg/sdk/stages_gen_test.go

```diff
@@ -0,0 +1,718 @@
+package sdk
+
+import "testing"
+
+func TestStages_CreateInternal(t *testing.T) {
+	id := RandomSchemaObjectIdentifier()
+
+	// Minimal valid CreateInternalStageOptions
+	defaultOpts := func() *CreateInternalStageOptions {
+		return &CreateInternalStageOptions{
+			name: id,
+		}
+	}
+
+	t.Run("validation: nil options", func(t *testing.T) {
+		var opts *CreateInternalStageOptions = nil
+		assertOptsInvalidJoinedErrors(t, opts, ErrNilOptions)
+	})
+
+	t.Run("validation: conflicting fields for [opts.OrReplace opts.IfNotExists]", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.OrReplace = Bool(true)
+		opts.IfNotExists = Bool(true)
+		assertOptsInvalidJoinedErrors(t, opts, errOneOf("CreateInternalStageOptions", "OrReplace", "IfNotExists"))
+	})
+
+	t.Run("basic", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.OrReplace = Bool(true)
+		opts.Temporary = Bool(true)
+		opts.FileFormat = &StageFileFormat{
+			Type: &FileFormatTypeCSV,
+		}
+		opts.Comment = String("some comment")
+		assertOptsValidAndSQLEquals(t, opts, "CREATE OR REPLACE TEMPORARY STAGE %s FILE_FORMAT = (TYPE = CSV) COMMENT = 'some comment'", id.FullyQualifiedName())
+	})
+
+	t.Run("all options", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.Temporary = Bool(true)
+		opts.IfNotExists = Bool(true)
+		opts.Encryption = &InternalStageEncryption{
+			Type: &InternalStageEncryptionFull,
+		}
+		opts.DirectoryTableOptions = &InternalDirectoryTableOptions{
+			Enable:          Bool(true),
+			RefreshOnCreate: Bool(true),
+		}
+		opts.FileFormat = &StageFileFormat{
+			FormatName: String("format name"),
+		}
+		opts.CopyOptions = &StageCopyOptions{
```

Comment:
please add more than one copy option (here and in other places), to see that we are generating multiple options correctly according to syntax

Reply:
Added